src/RefinedAst.hs

```diff
@@ -13,24 +13,27 @@ import Data.Map.Strict    (Map)
 import Data.List.NonEmpty hiding (fromList)
 import Data.ByteString       (ByteString)
 
-import Syntax
+import Syntax hiding (Invariants)
 import Data.Aeson
 import Data.Aeson.Types
 import Data.Vector (fromList)
 
 -- AST post typechecking
-data Behaviour = Behaviour
-  {_name :: Id,
-   _mode :: Mode,
-   _creation :: Bool,
-   _contract :: Id,
-   _interface :: Interface,
-   _preconditions :: Exp Bool,
-   _postconditions :: Exp Bool,
-   _contracts :: [Id], -- can maybe be removed; should be equivalent to Map.keys(_stateupdates)
-   _stateUpdates :: Map Id [Either StorageLocation StorageUpdate],
-   _returns :: Maybe ReturnExp
-  }
+data Behaviour =
+  Invariants [Exp Bool] [Id]
```

Comment:
@MrChico this is a bit messy because there is already an `Invariants` type in `Syntax.hs`. Do you have any suggestions for better naming / types here?

Reply:
Hmm, I think we might wanna lift `Invariants` to their own type, and let `Behaviour` just consist of `Behaviour` here. They could both be contained in a larger type, like `data Claim = B Behaviour | I Invariant` or `data Spec = Function Behaviour | Contract Invariant` or something like this. I'd also probably do `data Invariant = Invariant Id Exp Bool`, so we can tackle different claims individually.